Containerfile

```diff
@@ -23,21 +23,23 @@ RUN wget https://sjtodd.github.io/ling110/srilm-1.7.3.tar.gz && \
     make MAKE_PIC=yes World && \
     make cleanest 
 
-RUN pip install gensim \
+RUN mamba install -y \
```

Comment:
The `spacy` in mamba is incompatible with numpy>=2 

```
The following packages are incompatible
├─ jsonschema is not installable because it conflicts with any installable versions previously reported;
├─ numpy is not installable because it conflicts with any installable versions previously reported;
├─ python is not installable because it conflicts with any installable versions previously reported;
└─ spacy is not installable because there are no viable options
   ├─ spacy [1.1.0|1.2.0|...|1.9.0] would require
   │  └─ numpy 1.11* , which conflicts with any installable versions previously reported;
   ├─ spacy [1.1.0|1.2.0|...|1.9.0] would require
   │  └─ numpy 1.12* , which conflicts with any installable versions previously reported;
   ├─ spacy [2.0.10|2.0.11|...|2.0.9] would require
   │  └─ python 2.7* , which conflicts with any installable versions previously reported;
   ├─ spacy [2.0.10|2.0.11|...|2.0.9] would require
   │  └─ python 3.5* , which conflicts with any installable versions previously reported;
   ├─ spacy [2.0.10|2.0.11|...|2.0.9] would require
   │  └─ python 3.6* , which conflicts with any installable versions previously reported;
   ├─ spacy [2.0.11|2.0.12|2.0.13] would require
   │  └─ numpy >=1.9.3,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.0.16|2.0.17|...|2.2.2] would require
   │  └─ numpy >=1.15.4,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy 2.0.18 would require
   │  └─ numpy >=1.15.1,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy 2.0.18 would require
   │  └─ numpy >=1.15.0,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.1.1|2.1.2|...|2.1.6] would require
   │  └─ jsonschema >=2.6.0,<3.0.0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.2.3|2.2.4|...|2.3.4] would require
   │  └─ numpy >=1.16.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.3.2|2.3.3|2.3.4|2.3.5] would require
   │  └─ numpy >=1.19.4,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.3.5|3.0.0|...|3.0.5] would require
   │  └─ numpy >=1.16.6,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.3.5|2.3.7|...|3.4.1] would require
   │  └─ numpy >=1.19.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [2.3.7|3.0.6|3.1.0|3.1.1] would require
   │  └─ numpy >=1.17.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.1.2|3.1.3|...|3.2.2] would require
   │  └─ numpy >=1.18.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.2.1|3.2.2|3.2.4] would require
   │  └─ numpy >=1.21.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.2.3|3.3.0|...|3.6.1] would require
   │  └─ numpy >=1.21.6,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.4.2|3.4.3|3.4.4|3.5.0|3.5.1] would require
   │  └─ numpy >=1.20.3,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy 3.4.3 would require
   │  └─ numpy >=1.23.4,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.4.4|3.5.0|...|3.7.6] would require
   │  └─ numpy >=1.23.5,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy [3.7.1|3.7.2|...|3.7.6] would require
   │  └─ numpy >=1.22.4,<2.0a0 , which conflicts with any installable versions previously reported;
   ├─ spacy 3.7.5 would require
   │  └─ numpy >=1.26.4,<2.0a0 , which conflicts with any installable versions previously reported;
   └─ spacy 3.7.5 would require
      └─ numpy >=1.24.4,<2.0a0 , which conflicts with any installable versions previously reported.
```

Reply:
It looks like space 3.8 is compatible with numpy 2, but the latest in mamba is currently 3.7.5.

https://github.com/explosion/spaCy/releases/tag/release-v3.8.2

The good news is Pypi has v3.8, so installing spaCy from pip after the mamba call should work.

https://pypi.org/project/spacy/

Reply:
There's a decision to be made - keep gensim and downgrade numpy to 1.26.4 or keep Numpy 2.1 and don't ship gensim.